align/compiler/match_graph.py

```diff
@@ -77,6 +86,7 @@ def annotate(self):
                     if isinstance(symm_blocks, dict) and "graph" in symm_blocks.keys():
                         logger.debug(f"added new hierarchy: {symm_blocks['name']} {symm_blocks['graph'].nodes()}")
                         self.hier_graph_dict[symm_blocks['name']] = symm_blocks
+                        assert False, "Don't understand what's being deleted here"
```

Comment:
Also wanted you to look at this bit @kkunal1408. I inserted an assertion because the deletion didn't make sense to me but it was never triggered.